pkg/vault/contracts/VaultFactory.sol

```diff
@@ -2,28 +2,37 @@
 
 pragma solidity ^0.8.24;
 
-import { Authentication } from "@balancer-labs/v3-solidity-utils/contracts/helpers/Authentication.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { Ownable2Step } from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import { Create2 } from "@openzeppelin/contracts/utils/Create2.sol";
+
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
 import { IAuthorizer } from "@balancer-labs/v3-interfaces/contracts/vault/IAuthorizer.sol";
+import {
+    ReentrancyGuardTransient
+} from "@balancer-labs/v3-solidity-utils/contracts/openzeppelin/ReentrancyGuardTransient.sol";
 import { CREATE3 } from "@balancer-labs/v3-solidity-utils/contracts/solmate/CREATE3.sol";
-import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
-import { Create2 } from "@openzeppelin/contracts/utils/Create2.sol";
 
 import { ProtocolFeeController } from "./ProtocolFeeController.sol";
 import { VaultExtension } from "./VaultExtension.sol";
 import { VaultAdmin } from "./VaultAdmin.sol";
 
 /// @notice One-off factory to deploy the Vault at a specific address.
-contract VaultFactory is Authentication {
+contract VaultFactory is ReentrancyGuardTransient, Ownable2Step {
```

Comment:
What are the advantages as opposed to Authentication?

Reply:
Ownable2Step makes sense here, in case the account gets compromised. We'd need to use this again if we ever had to redeploy the Vault.

Reply:
it can't be redeployed with the flag anyways. We'd need to redeploy the factory in that case.
We can change the flag to a mapping (salt => boolean) if that was the case. But I think the current approach is fine.

> What are the advantages as opposed to Authentication?

Authentication needs governance to grant the permission to deploy, which makes it operationally more cumbersome.

The deployment depends on the deployer account anyways, which is the one that generates the right address for the vault factory, and in turn generates the right vault address given the correct salt. So now we're saying that besides deploying the factory, the deployer also needs to call `create`.